File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        white=blocks[0:k].count('W')
        black=k-white

        r=k
        res=white
        for l in blocks[0:len(blocks)-k]:
            if l=='B':
                black-=1
            elif l=='W':
                white-=1
            
            if blocks[r]=='B':
                black+=1
            elif blocks[r]=='W':
                white+=1
            r+=1
            res=min(white,res)
        return res

            

                 

        
        

File: 2379-minimum-recolors-to-get-k-consecutive-black-blocks/2379-minimum-recolors-to-get-k-consecutive-black-blocks.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int white=0;
        for(int i=0;i<k;i++){
            if(blocks.charAt(i)=='W')
              white++;
            }
        int r=k;
        int res=white;
        for(int i=0;i<blocks.length()-k;i++){
            if(blocks.charAt(i)=='W'){
                white--;
            }

            if(blocks.charAt(r)=='W'){
                white++;
            }
            r++;

            res=Math.min(white,res);

            
        }
        return res;}
        
    }
